### ISSUE TITLE
[Bug] - Doing a fresh install on a fresh install of Steam OS.does not add shortcuts to the non steam section

### ISSUE THREAD

**Issue by PS2ClassicsVault** [moraroy/NonSteamLaunchersDecky#31]
When installing launchers from the plugin via decky loader, the launchers will install but not add short cuts to the non steam area in game mode

**Comment by moraroy** [comment]
@PS2ClassicsVault good catch, even when restarting steam?

**Comment by PS2ClassicsVault** [comment]
> @PS2ClassicsVault good catch, even when restarting steam?

Yup when I restart steam still doesn't add them,  or show up I'm on the latest.

I think this became an issue when steam os updated to v3.6.19

**Comment by moraroy** [comment]
@PS2ClassicsVault I haven't reset my deck yet,I plan to though, go to desktop mode and add a random non steam item, like chrome or something using "add to steam" jump back into game mode and run a scan, I think maybe then it will add. Let me know if that works

**Comment by PS2ClassicsVault** [comment]
> @PS2ClassicsVault I haven't reset my deck yet,I plan to though, go to desktop mode and add a random non steam item, like chrome or something using "add to steam" jump back into game mode and run a scan, I think maybe then it will add. Let me know if that works

Okay just did what you told me and then tried doing a manual scan in the plugin but the manual scan seems to be broken as well when I click it it should of done it's job and be on a cool down before I could do it again, but does nothing I even tried installing the launchers again and they install really fast for some reason and still doesn't add any of the launchers to game mode

**Comment by moraroy** [comment]
@PS2ClassicsVault yikes that's not good. Run the NSLPlugin.desktop again this will install a fresh copy if that still is happening I goofed somewhere m

**Comment by PS2ClassicsVault** [comment]
> @PS2ClassicsVault yikes that's not good. Run the NSLPlugin.desktop again this will install a fresh copy if that still is happening I goofed somewhere m

Alright I re ran the installer and it did what you said but the issue is still present.

**Comment by moraroy** [comment]
@PS2ClassicsVault dang? And this is on a fresh os install, with decky loader, and the NSLPlugin.desktop? Something is definitely not right, go into the logs "/home/deck/homebrew/logs/NonSteamLaunchers/" it should give you clues on what's happening if there is any red or white let me know , it should be all green

**Comment by moraroy** [comment]
@PS2ClassicsVault I believe it's fixed give it a try with the .desktop file, I believe I messed up sorry about that

**Comment by PS2ClassicsVault** [comment]
> @PS2ClassicsVault I believe it's fixed give it a try with the .desktop file, I believe I messed up sorry about that

I gave it another try this time it did do as it was supposed to, but still didn't add them to steam, I tried scanning hit it was an infinite scan.

I even tried the streaming sites as well and they were starting doing nothing except saying installing, I included all logs and attached them to here hopefully that helps

[NonSteamLaunchers.zip](https://github.com/user-attachments/files/17573958/NonSteamLaunchers.zip)

**Comment by moraroy** [comment]
@PS2ClassicsVault yes that was very helpful thank you, I didn't see too many errors except one but let me know if the latest version fixes your issue, just to let you know, the script needs to be ran once in order for the env_vars to populate meaning you would need to install a launcher first and then attempt to do the streaming sites or custom sites.

**Comment by PS2ClassicsVault** [comment]
> @PS2ClassicsVault yes that was very helpful thank you, I didn't see too many errors except one but let me know if the latest version fixes your issue, just to let you know, the script needs to be ran once in order for the env_vars to populate meaning you would need to install a launcher first and then attempt to do the streaming sites or custom sites.

The launchers installed correctly and they were successfully added to game mode, and also how long should the streaming sites install should take?

**Comment by moraroy** [comment]
@PS2ClassicsVault it shouldn't take that long because it's going to first install chrome if you don't have it, since you did a fresh install im assuming it was trying to install? But make sure autoscan is on for those streaming sites.

**Comment by PS2ClassicsVault** [comment]
> @PS2ClassicsVault it shouldn't take that long because it's going to first install chrome if you don't have it, since you did a fresh install im assuming it was trying to install? But make sure autoscan is on for those streaming sites.

I do have chime already installed

**Comment by moraroy** [comment]
@AloushMan yep still working on this, give me some time to fix, once you had then installed did you scan for them using the plugin or the desktop version?

**Comment by AloushMan** [comment]
@moraroy no worries! 

So, when i input my sudo password in desktop mode, at the bottom right i get some "error pop up for a split second" and then says, installing.. and then going into gamemode. 

I actually did not try to scan on steam from desktop mode. I would click the launchers i want in gamemode, hit install and then try to do a manual scan there.

**Comment by moraroy** [comment]
@AloushMan @PS2ClassicsVault i think this may have been fixed recently tbh will need to test on a fresh os

**Comment by moraroy** [comment]
@AloushMan fixed in the latest version I think
